crates/sui-rosetta/src/construction.rs

```diff
@@ -252,11 +253,31 @@ pub async fn metadata(
     // make sure it works over epoch changes
     gas_price += 100;
 
+    if let InternalOperation::PaySui {
+        sender,
+        recipients,
+        amounts,
+    } = option.internal_operation
+    {
+        let metadata = pay_sui_to_metadata(
+            &context.client,
+            Some(gas_price),
+            sender,
+            recipients.clone(),
+            amounts,
+            budget,
+        )
+        .await?;
+        let budget = metadata.budget;
+        return Ok(ConstructionMetadataResponse {
+            metadata,
+            suggested_fee: vec![Amount::new(budget as i128, None)],
+        });
+    };
     // Get amount, objects, for the operation
     let (total_required_amount, objects) = match &option.internal_operation {
-        InternalOperation::PaySui { amounts, .. } => {
-            let amount = amounts.iter().sum::<u64>();
-            (Some(amount), vec![])
+        InternalOperation::PaySui { .. } => {
+            unreachable!("PaySui is already handled explicitly")
         }
         InternalOperation::PayCoin { amounts, .. } => {
```

Comment:
I guess pay-coin should be next to handled as there is currently no work-around.
eg. for stake, one can send sui to the sender (recipient = sender) to smash multiple sui-coins, before the stake operation, but I don't know if this would work for pay-coin.